docs/06_how-to-guides/40_multi-index/how-to-iterate-and-retrieve-a-multi_index-table.md

```diff
@@ -1,69 +1,49 @@
+## Overview
 
-For exemplification define the multi index contract definition like below:
+This guide provides instructions to iterate and retrieve data from a multi-index table.
 
-__multi_index_example.hpp__
-```cpp
-#include <eosio/eosio.hpp>
-using namespace eosio;
+## Reference
 
-// multi index example contract class
-class [[eosio::contract]] multi_index_example : public contract {
-   public:
-      using contract::contract;
+See the following code reference:
 
-      // contract class constructor
-      multi_index_example( name receiver, name code, datastream<const char*> ds ) :
-         // contract base class contructor
-         contract(receiver, code, ds),
-         // instantiate multi index instance as data member (find it defined below)
-         testtab(receiver, receiver.value)
-         { }
+* The [`multi-index`](../../classeosio_1_1multi__index) class.
+* The [`multi-index::find(...)`](../../group__multiindex#function-find) method.
 
-      // the row structure of the multi index table, that is, each row of the table
-      // will contain an instance of this type of structure
-      struct [[eosio::table]] test_table {
-        // this field stores a name for each row of the multi index table
-        name test_primary;
-        // additional data stored in table row
-        uint64_t datum;
-        // mandatory definition for primary key getter
-        uint64_t primary_key( ) const { return test_primary.value; }
-      };
+## Before you begin
 
-      // the multi index type definition, for ease of use define a type alias `test_tables`, 
-      // based on the multi_index template type, parametarized with a random name and 
-      // the test_table data structure
-      typedef eosio::multi_index<"testtaba"_n, test_table> test_tables;
+Make sure you have the following prerequisites in place:
 
-      // the multi index table instance declared as a data member of type test_tables
-      test_tables testtab;
+* An EOSIO development environment, for details consult the [Get Started Guide](https://developers.eos.io/welcome/latest/getting-started-guide/index),
+* A multi-index `testab` table instance which stores `user` objects indexed by the primary key which is of type `eosio::name`. Consult the section [How to instantiate a multi-index table](./how-to-instantiate-a-multi-index-table) to learn how to set it up.
 
-      [[eosio::action]] void set( name user );
+## Procedure
 
-      using set_action = action_wrapper<"set"_n, &multi_index_example::set>;
-};
-```
+Complete the following steps to iterate, retrieve and print data from the `testtab` multi-index table.
 
-The steps below show how to iterate and retrieve a multi index table.
+### 1. Define The print(...) Action
 
-1. Add to the above multi index example contract an action `print` which gets as parameter an acount name
+Add to the definition of the `testtab` multi-index table the `print` action which gets as parameter an account name.
```

Comment:
maybe:
Add a `print` action to the `testtab` multi-index table. The `print` action takes an account name as a parameter.